src/physical/tabular/operations/triescan_select.rs

```diff
@@ -225,13 +226,31 @@ impl<'a> TrieScanSelectValue<'a> {
             macro_rules! init_scans_for_datatype {
                 ($variant:ident) => {
                     unsafe {
-                        let value = if let StorageValueT::$variant(value) =
-                            assignemnt.value.to_storage_value(dict)
+                        let lower = if let Some(StorageValueT::$variant(value)) = assignemnt
+                            .interval
+                            .lower
+                            .bound()
+                            .map(|l| l.to_storage_value(dict))
                         {
-                            value
+                            Some(value)
                         } else {
-                            panic!("Expected a column scan of type {}", stringify!($variant));
+                            None
+                        };
+
+                        let upper = if let Some(StorageValueT::$variant(value)) = assignemnt
+                            .interval
+                            .upper
+                            .bound()
+                            .map(|l| l.to_storage_value(dict))
+                        {
+                            Some(value)
+                        } else {
+                            None
                         };
+                        let interval = Interval::new(
+                            Self::translate_interval_bound(&assignemnt.interval.lower, lower),
```

Comment:
Why is the actual translation not part of this function?

Reply:
This was the idea of the function, i.e. it should translate `Interval<DatatypeT>` to `Interval<T>`, but this turned out to be more difficult because `to_storage_value` returns a `StorageTypeT`. Now to get a `T` I would have to do a `if let StorageTypeT::$variant(...) = ...` but I cannot have the `$variant` in the translate function.

I dont't know if there is a better solution

Reply:
I kind of understand the problem but now the function just wraps its second parameter into the enum variant kind of given by the first parameter. This does not really match the name of the function. I think this should be rewritten somehow but I'm not sure how without fiddling around with it. Maybe having additional macros to integrate the function into the existing macro would be a possible approach...

Reply:
Looks much better now, thanks!